invenio_oauth2server/ext.py

```diff
@@ -186,6 +188,7 @@ def verify_oauth_token_and_set_current_user():
 
         if valid:
             request.oauth = req
+        request.oauth_verify_run = True
```

Comment:
```suggestion
    request.oauth_verify_run = True
```

Maybe it's more generic this way